pkg/config/credentials.go

```diff
@@ -0,0 +1,127 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/docker/distribution/registry/client/auth"
+	"github.com/docker/distribution/registry/client/transport"
+	"github.com/openshift/library-go/pkg/image/registryclient"
+	"github.com/openshift/oc/pkg/helpers/image/credentialprovider"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+// New creates a new credential store for the provided secret content
+func New(secret []byte) (auth.CredentialStore, error) {
+
+	cfg, err := readDockerConfigJSONFileFromBytes(secret)
+	if err != nil {
+		return nil, err
+	}
+	keyring := &credentialprovider.BasicDockerKeyring{}
+	keyring.Add(cfg)
+	return &keyringCredentialStore{
+		DockerKeyring:     keyring,
+		RefreshTokenStore: registryclient.NewRefreshTokenStore(),
+	}, nil
+}
+
+// CreateContext a new context for the registryClient of `oc mirror`
+func CreateContext(secret []byte, skipVerification bool) (*registryclient.Context, error) {
+	rt, err := rest.TransportFor(&rest.Config{})
+	if err != nil {
+		return nil, err
+	}
+	insecureRT, err := rest.TransportFor(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}})
```

Comment:
TLS verification should only be skipped when `--skip-tls` is set right?

Reply:
Good catch. Yes, you are right. Updating now.